chain/chain/src/types.rs

```diff
@@ -633,7 +636,9 @@ pub trait RuntimeAdapter: Send + Sync {
             is_new_chunk,
             is_first_block_with_chunk_of_version,
             state_patch,
-        )
+        );
+        timer.observe_duration();
```

Comment:
I'd skip the `observe_duration()` call in favor of `drop()` recording the duration.